src/utils/events.ts

```diff
@@ -0,0 +1,183 @@
+import { DomainColor } from '~gql';
+import { Domain, Falsy } from '~types';
+import { convertToCapitalized } from './strings';
+
+interface ColonyMetadataChanges {
+  nameChanged: boolean;
+  logoChanged: boolean;
+  tokensChanged: boolean;
+  verifiedAddressesChanged: boolean;
+}
+
+type DomainMetadata = Pick<Domain, 'color' | 'name' | 'description'>;
+
+export const getColonyMetadataMessageValues = (
```

Comment:
Dynamically generating the message text according to the values that changed

Reply:
As I mentioned previously, we should store metadata history in an array format, so this will end up having to account for that

Reply:
@jakubcolony I have left a comment in the code but just pinging you here as well since I imagine you'll be looking after this